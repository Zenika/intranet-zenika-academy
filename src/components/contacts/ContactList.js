import React from 'react';
import './ContactList.scss';
import Card from '../card/Card';

const data = [
  {
    id: 535345554,
    name: 'John Smith',
    description: 'Sunt magna duis proident in exercitation exercitation sit esse magna.',
  },
  {
    id: 645964055,
    name: 'GÃ©rard Martin',
    description: 'Proident adipisicing proident officia dolor mollit proident aliqua.',
  },
  {
    id: 685968406,
    name: 'Sylvie Lafite',
    description: 'Qui eiusmod ut cupidatat elit veniam duis. Nisi quis sit deserunt reprehenderit veniam aute consectetur.',
  },
];

export function generateCards() {
  const cards = [];
  data.forEach((person) => {
    cards.push(<Card key={person.id} data={person} />);
  });
  return cards;
}

const ContactList = () => (
  <>
    <h1 className="title is-2 mbmd">Contacts utiles</h1>
    <div className="ContactListCardContainer">
      {generateCards()}
    </div>
  </>
);

export default ContactList;
