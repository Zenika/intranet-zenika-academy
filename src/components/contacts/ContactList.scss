.row{
    /* background-color: red; */
    display:flex;
    flex-direction: row;
    margin:4rem 0;
    justify-content: center;
    align-items: center;
}

.photo{
    /* background-color: blue; */
    border: 1px black solid;
    border-radius: 80px;
    display: flex;
    justify-content: center;
    align-items: center;
    height: 3.5rem;
    width: 3.5rem;
    margin:4rem;
    padding: 0.5rem;
}

.textBox{
    /* background-color: green; */
    border:1px solid black;
    padding:0.3rem;
    flex:1;
    height: 11rem;
}

.description{
    margin-top:0.5rem;
}
