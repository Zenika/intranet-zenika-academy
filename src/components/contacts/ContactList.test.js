import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ContactList from './ContactList';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<ContactList />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe('ContactList', () => {
  it('renders', () => {
    const component = renderer.create(<ContactList />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
