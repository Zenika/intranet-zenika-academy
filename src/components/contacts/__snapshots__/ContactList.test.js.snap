// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactList renders 1`] = `
Array [
  <h1
    className="title is-2 mbmd"
  >
    Contacts utiles
  </h1>,
  <div
    className="ContactListCardContainer"
  >
    <article
      className="card"
    >
      <section
        className="image is-4by3 card-image"
      >
        <img
          alt="photo de contact"
          src="https://bulma.io/images/placeholders/1280x960.png"
        />
      </section>
      <section
        className="card-content"
      >
        <div
          className="media"
        >
          <div
            className="media-content"
          >
            <h1
              className="title is-5"
            >
              John Smith
            </h1>
          </div>
        </div>
        <h2
          className="subtitle is-6"
        >
          Descriptif de la personne et contacts réseaux / autres
        </h2>
        <p
          className="content"
        >
          Sunt magna duis proident in exercitation exercitation sit esse magna.
        </p>
      </section>
    </article>
    <article
      className="card"
    >
      <section
        className="image is-4by3 card-image"
      >
        <img
          alt="photo de contact"
          src="https://bulma.io/images/placeholders/1280x960.png"
        />
      </section>
      <section
        className="card-content"
      >
        <div
          className="media"
        >
          <div
            className="media-content"
          >
            <h1
              className="title is-5"
            >
              Gérard Martin
            </h1>
          </div>
        </div>
        <h2
          className="subtitle is-6"
        >
          Descriptif de la personne et contacts réseaux / autres
        </h2>
        <p
          className="content"
        >
          Proident adipisicing proident officia dolor mollit proident aliqua.
        </p>
      </section>
    </article>
    <article
      className="card"
    >
      <section
        className="image is-4by3 card-image"
      >
        <img
          alt="photo de contact"
          src="https://bulma.io/images/placeholders/1280x960.png"
        />
      </section>
      <section
        className="card-content"
      >
        <div
          className="media"
        >
          <div
            className="media-content"
          >
            <h1
              className="title is-5"
            >
              Sylvie Lafite
            </h1>
          </div>
        </div>
        <h2
          className="subtitle is-6"
        >
          Descriptif de la personne et contacts réseaux / autres
        </h2>
        <p
          className="content"
        >
          Qui eiusmod ut cupidatat elit veniam duis. Nisi quis sit deserunt reprehenderit veniam aute consectetur.
        </p>
      </section>
    </article>
  </div>,
]
`;
